### DISCUSSION TITLE
How to downsample counts?

### DISCUSSION THREAD

**Discussion by Zig-E-Stardust** [satijalab/seurat#7163] [Q&A]
Hey there!
I've been having a recurring problem (so far happening with 4 unrelated datasets/projects): when looking for DEGs between, for instance, a WT group and a KO group, the group that has a higher "nCount" (which corresponds with a higher "nFeature") always has a substantially higher number of detected DEGs, no matter which method I'm using to detect DEGs and regardless of whether I'm using the "RNA" assay or the "SCT" assay (in both cases, I use the "data" slot, but the same problem occurs also when using the "counts" slot for the SCT assay).

I assume the problem stems from the inter-group difference in the number of detected transcripts which, in turn, I strongly suspect is due to a technical artifact. Therefore, I want to try to downsample the counts of the group with the higher "nCount" to fit the counts of the group with the lower "nCount". I've noticed that Seurat offers a way to do that, with the "SampleUMI" function, but I have a few questions:
1. Do you think that's a suitable way to do that? If not, can you please offer an alternative?
2. At which point in the analysis "pipeline" should I do that? I've already done the QC, integration, clustering, cluster annotation etc., so, naturally, I'd be happy if I can do that "count downsampling" after all that (at least as a "draft", to see if it can help solve the problem), but... Can I? Or should I do it at some earlier step (e.g. before integration)?
3. How should I choose the value for the "max.umi" parameter? For instance, I could set it to the median of the "nCount" of the group (or, perhaps better, the mouse [they're hashtagged]) with the lowest "nCount", but then half of the cells in that group would still have a lower "nCount" than chosen "max.umi", while, in the other group, the proportion of cells with "nCount"<"max.umi" will be smaller.

Thanks!
Omer

**Comment by jesspei** [comment]
Hi Omer, I am having the exact same issue and was wondering what you did to resolve it? 

Thank you! 
Jess
